packages/firestore/src/local/indexeddb_remote_document_cache.ts

```diff
@@ -102,11 +104,17 @@ class IndexedDbRemoteDocumentCacheImpl implements IndexedDbRemoteDocumentCache {
    */
   removeEntry(
     transaction: PersistenceTransaction,
-    documentKey: DocumentKey
+    documentKey: DocumentKey,
+    readTime: SnapshotVersion
   ): PersistencePromise<void> {
     const store = remoteDocumentsStore(transaction);
-    const key = dbKey(documentKey);
-    return store.delete(key);
+    const path = documentKey.path.toArray();
+    return store.delete([
+      path.slice(0, path.length - 2),
```

Comment:
There are quite some places using `length - n` to get what we want, maybe we should put all these in `DocumentKey` instead.